gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Policy.java

```diff
@@ -0,0 +1,389 @@
+      }
+      this.members = membersListBuilder.build();
+    }
+
+    public static Binding binding(RoleType role, List<Member> members) {
+      return new Binding(role, members);
+    }
+
+    public RoleType role() {
+      return role;
+    }
+
+    public List<Member> members() {
+      return members;
+    }
+
+    public static Builder builder() {
+      return new Builder();
+    }
+
+    public Builder toBuilder() {
+      List<Member> mutableMembers = new ArrayList<>();
+      for (Member member : members) {
+        mutableMembers.add(member);
+      }
+      return new Builder().role(role).members(mutableMembers);
+    }
+
+    com.google.api.services.cloudresourcemanager.model.Binding toPb() {
+      com.google.api.services.cloudresourcemanager.model.Binding bindingPb =
+          new com.google.api.services.cloudresourcemanager.model.Binding();
+      if (role != null) {
+        bindingPb.setRole("roles/" + role.toString().toLowerCase());
+      }
+      List<String> membersPb = new ArrayList<>(members.size());
+      for (Member member : members) {
+        if (member.emailOrDomain() != null) {
+          membersPb.add(member.type().prefix() + member.emailOrDomain());
+        } else {
+          membersPb.add(member.type().prefix());
+        }
+      }
+      bindingPb.setMembers(membersPb);
+      return bindingPb;
+    }
+
+    static Binding fromPb(com.google.api.services.cloudresourcemanager.model.Binding bindingPb) {
+      RoleType role =
+          (bindingPb.getRole() == null)
+              ? null : RoleType.valueOf(bindingPb.getRole().split("/")[1].toUpperCase());
+      List<Member> members = new ArrayList<>();
+      if (bindingPb.getMembers() != null) {
+        for (String memberPb : bindingPb.getMembers()) {
+          String[] memberInfo = memberPb.split(":", 2);
+          String memberTypeStr = memberInfo[0];
+          String emailOrDomain = (memberInfo.length > 1) ? emailOrDomain = memberInfo[1] : null;
+          switch (memberTypeStr) {
+            case "allUsers":
+              members.add(new Member(MemberType.ALL_USERS, null));
+              break;
+            case "allAuthenticatedUsers":
+              members.add(new Member(MemberType.ALL_AUTHENTICATED_USERS, null));
+              break;
+            case "user":
+              members.add(new Member(MemberType.USER, checkNotNull(emailOrDomain)));
+              break;
+            case "serviceAccount":
+              members.add(new Member(MemberType.SERVICE_ACCOUNT, checkNotNull(emailOrDomain)));
+              break;
+            case "group":
+              members.add(new Member(MemberType.GROUP, checkNotNull(emailOrDomain)));
+              break;
+            case "domain":
+              members.add(new Member(MemberType.DOMAIN, checkNotNull(emailOrDomain)));
+              break;
+            default:
+              throw new UnsupportedOperationException("Unsupported member type: " + memberTypeStr);
+          }
```

Comment:
I feel like this logic should be moved to a `Member.fromPb(String memberPb)` method, what do you think?